packages/oauth/oauth_common.js

```diff
@@ -6,6 +6,8 @@ OAuth._redirectUri = function (serviceName, config, params, absoluteUrlOptions)
   // detect whether we need to be backwards compatible by checking for
   // the absence of the `loginStyle` field, which wasn't used in the
   // code which had the "?close" argument.
+  // This logic is duplicated in the tool so that the tool can do OAuth
```

Comment:
Yeah, I think we should back this out. I'll try to write up a coherent explanation.